source/Symbol/SwiftASTContext.cpp

```diff
@@ -5129,12 +5129,12 @@ SwiftASTContext::GetReferentType(const CompilerType &compiler_type) {
 
   if (compiler_type.IsValid() &&
       llvm::dyn_cast_or_null<SwiftASTContext>(compiler_type.GetTypeSystem())) {
-    swift::CanType swift_can_type(GetCanonicalSwiftType(compiler_type));
-    swift::TypeBase *swift_type = swift_can_type.getPointer();
-    if (swift_type && llvm::isa<swift::WeakStorageType>(swift_type))
+    swift::Type swift_type(GetSwiftType(compiler_type));
+    swift::TypeBase *swift_typebase = swift_type.getPointer();
+    if (swift_type && llvm::isa<swift::WeakStorageType>(swift_typebase))
```

Comment:
`if (llvm::dyn_cast_or_null<swift::WeakStorageType>(GetSwiftType(compiler_type).getPointer())` ?

Reply:
Sure, I can do this cleanup as a follow up.